### ISSUE TITLE
Twine hangs without prompting for password, no uploads to PyPI happen

### ISSUE THREAD

**Issue by monocongo** [pypa/packaging-problems#197]
Thank you for providing feedback on Python packaging!

To help us help you, please fill out as much of the following as you can. If a question is not relevant, feel free to skip it.

1. What is your operating system and version?

Windows 10

2. What is your Python version?

$ python --version
Python 3.7.0

3. What version of pip do you have?

$ pip --version
pip 10.0.1 from C:\home\miniconda3\envs\packaging\lib\site-packages\pip (python 3.7)

4. Could you describe your issue in as much detail as possible?

I am trying to upload my tarball and wheel files using twine but it hangs after I enter my username, with no prompt for my password.

I am on a Windows 10 system, running the bash terminal that came included with the Windows git client, using Minoconda3 for my Python distribution and performing this in a conda environment that has twine installed.

$ twine --version
twine version 1.11.0 (pkginfo: 1.4.2, requests: 2.19.1, setuptools: 40.2.0,
requests-toolbelt: 0.8.0, tqdm: 4.25.0)

$ twine upload --repository-url https://test.pypi.org/legacy/ dist/*
Uploading distributions to https://test.pypi.org/legacy/
Enter your username: monocongo
<THIS IS AS FAR AS IT GETS, NO PROMPT FOR PASSWORD AS EXPECTED>

I see no errors, maybe someone can advise as to where I can monitor things to detect what's causing the issue?

Thanks in advance for any suggestions!

**Comment by benoit-pierre** [comment]
Does it happen if you run twine from a regular command prompt (cmd)?

As a workaround, you could provide your authentication info in `~/.pypirc` (that's on Unix, not sure what the exact path would be on Windows):
```
[pypi]
username = user
password = pass
```

**Comment by monocongo** [comment]
Thanks a lot for these suggestions, Benoit.

Using the regular command prompt (DOS console) worked for me.

Does this indicate a bug of some sort, in terms of the interaction of twine with the git/bash shell console for Windows?

In any event thanks again for your help, it really got me over a hump.

**Comment by benoit-pierre** [comment]
@monocongo: see https://github.com/mintty/mintty/issues/481

**Comment by benoit-pierre** [comment]
The Git for Windows release notes point to a possible workaround: https://github.com/git-for-windows/build-extra/blob/master/ReleaseNotes.md#known-issues

**Comment by ju-sh** [comment]
I'm getting similar problem now. The only difference is that the prompt for even the username is not showing up. It get stuck at the

    Uploading distributions to https://upload.pypi.org/legacy/

part itself.

I used the following command.

     python3 -m twine upload --repository-url https://upload.pypi.org/legacy/ dist/*

OS: Ubuntu 18.04
twine: 3.1.1 
pkginfo: 1.5.0.1 
requests: 2.23.0 
setuptools: 39.0.1,
requests-toolbelt: 0.9.1 
pip: 9.0.1

Is there any workaround?

I tried using a `~/.pypirc` as well without any change...
